idarling/core/hooks.py

```diff
@@ -525,15 +526,15 @@ def sgr_changed(self, start_ea, end_ea, regnum, value, old_value, tag):
     #     return 0
 
     def make_data(self, ea, flags, tid, size):
-        # TODO: tid --> struct name
         self._plugin.logger.debug("make_data(ea = %x, flags = %x, tid = %x, size = %x)" % (ea, flags, tid, size))
-        self._send_packet(evt.MakeDataEvent(ea, flags, size, tid))
+        self._send_packet(evt.MakeDataEvent(ea, flags, size, ida_struct.get_struc_name(tid) if tid != ida_idaapi.BADADDR else ''))
```

Comment:
I'd like to avoid sending an event if there is no tid with a struc_name == '' as we won't be able to replay it to other clients.

Do you remember when you have a tid == BADADDR so we can figure out why?

Reply:
tid can be BADADDR when something make data to Dword, Word, Byte or string for example. But need change BADADDR to BADNODE.

```
idaman bool ida_export create_data(
        ea_t ea,
        flags_t dataflag,
        asize_t size,
        tid_t tid);

```
/// Convert to data (byte, word, dword, etc).
/// This function may be used to create arrays.
/// \param ea        linear address
/// \param dataflag  type of data. Value of function byte_flag(), word_flag(), etc.
/// \param size      size of array in bytes. should be divisible by the size of
///                  one item of the specified type. for variable sized items
///                  it can be specified as 0, and the kernel will try to calculate the size.
/// \param tid       type id. If the specified type is a structure,
///                  then tid is structure id. Otherwise should be #BADNODE.
/// \return success

Reply:
Thanks for clarifying, makes sense